src/python/WMCore/BossAir/Plugins/SimpleCondorPlugin.py

```diff
@@ -0,0 +1,590 @@
+                (newStatus,location) = jobInfo[job['gridid']]
+
+            # check for status changes
+            if newStatus != job['status']:
+
+                # update location info for Idle->Running transition
+                if newStatus == '2' and job['status'] == '1':
+                    if location:
+                        job['location'] = location
+                        logging.debug("JobAdInfo: Job location for jobid=%i gridid=%s changed to %s", job['jobid'], job['gridid'], location)
+
+                job['status'] = newStatus
+                job['status_time'] = int(time.time())
+                logging.debug("JobAdInfo: Job status for jobid=%i gridid=%s changed to %s", job['jobid'], job['gridid'], job['status'])
+                changeList.append(job)
+
+            job['globalState'] = SimpleCondorPlugin.stateMap()[newStatus]
+
+            # stop tracking finished jobs
+            if job['globalState'] in [ 'Complete', 'Error' ]:
+                completeList.append(job)
+            else:
+                runningList.append(job)
+
+        logging.debug("SimpleCondorPlugin tracking : %i/%i/%i (Executing/Changing/Complete)",
+                      len(runningList), len(changeList), len(completeList))
+
+        return runningList, changeList, completeList
+
+    def complete(self, jobs):
+        """
+        Do any completion work required
+
+        In this case, look for a returned logfile
+        """
+
+        for job in jobs:
+
+            if job.get('cache_dir') is None or job.get('retry_count') is None:
+                # Then we can't do anything
+                logging.error("Can't find this job's cache_dir in SimpleCondorPlugin.complete")
+                logging.error("cache_dir: %s", job.get('cache_dir', 'Missing'))
+                logging.error("retry_count: %s", job.get('retry_count', 'Missing'))
+                continue
+
+            reportName = os.path.join(job['cache_dir'], 'Report.%i.pkl' % job['retry_count'])
+            if os.path.isfile(reportName) and os.path.getsize(reportName) > 0:
+                # everything in order, move on
+                continue
+            elif os.path.isdir(reportName):
+                # Then something weird has happened. Report error, do nothing
+                logging.error("The job report for job with id %s and gridid %s is a directory", job['id'], job['gridid'])
+                logging.error("Ignoring this, but this is very strange")
+            else:
+                logging.error("No job report for job with id %s and gridid %s", job['id'], job['gridid'])
+
+                if os.path.isfile(reportName):
+                    os.remove(reportName)
+
+                # create a report from scratch
+                condorReport = Report()
+                logOutput = 'Could not find jobReport\n'
+
+                if os.path.isdir(job['cache_dir']):
+                    # TODO: Alan - how about we look at the .out or .err file instead of the .log ?
```

Comment:
I think 50 line is fine. It shouldn't make a big difference unless we propagate whole log file.